### ISSUE TITLE
win_copy: MemoryError on huge files (+3GB)

### ISSUE THREAD

**Issue by dagwieers** [ansible/ansible#20023]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

win_copy

##### ANSIBLE VERSION

v2.2

##### OS / ENVIRONMENT

Windows 2012

##### SUMMARY

When doing a win_copy of a large ISO file (multiple files tested) of over 3GB in size, win_copy bails out with the below message:
```
TASK [windows : Transfer ISO image] **************************************
task path: /home/user/project/roles/windows/tasks/something.yml:21
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py", line 119, in run
    res = self._execute()
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_executor.py", line 490, in _execute
    result = self._handler.run(task_vars=variables)
  File "/usr/lib/python2.7/site-packages/ansible/plugins/action/copy.py", line 153, in run
    source_full = self._loader.get_real_file(source_full)
  File "/usr/lib/python2.7/site-packages/ansible/parsing/dataloader.py", line 402, in get_real_file
    if is_encrypted_file(f):
  File "/usr/lib/python2.7/site-packages/ansible/parsing/vault/__init__.py", line 152, in is_encrypted_file
    b_vaulttext = to_bytes(to_text(vaulttext, encoding='ascii', errors='strict'), encoding='ascii', errors='strict')
  File "/usr/lib/python2.7/site-packages/ansible/module_utils/_text.py", line 177, in to_text
    return obj.decode(encoding, errors)
MemoryError
 
fatal: [windows-2012]: FAILED! => {
    "failed": true,
    "msg": "Unexpected failure during module execution.",
    "stdout": ""
}
```

**Comment by dagwieers** [comment]
@abadger As requested.

**Comment by jhawkesworth** [comment]
It would be interesting to get an idea of the memory used.  I haven't looked at the code recently but there used to be a huge overhead in chopping up file contents and sending over web services interface.

**Comment by dagwieers** [comment]
Already fixed by #18368 !
